Sources/LeafKit/LeafLexer/LeafToken.swift

```diff
@@ -29,189 +140,66 @@ public enum Operator: String, Equatable, CustomStringConvertible {
     case greaterThanOrEquals = ">="
     case lessThan = "<"
     case lessThanOrEquals = "<="
-    
+
     case plus = "+"
     case minus = "-"
     case divide = "/"
     case multiply = "*"
-    
+
     case and = "&&"
     case or = "||"
     
+    /// Raw string value of the operator - eg `!=`
     public var description: String { return rawValue }
 }
 
-extension Operator {
-    var isBoolean: Bool {
-        switch self {
-            case .not,
-                 .equals,
-                 .notEquals,
-                 .greaterThan,
-                 .greaterThanOrEquals,
-                 .lessThan,
-                 .lessThanOrEquals,
-                 .and,
-                 .or:
-                return true
-            default:
-                return false
-        }
-    }
-    
-    var isBinary: Bool {
-        switch self {
-            case .not: return false
-            default: return true
-        }
-    }
-    
-    var isUnaryPrefix: Bool {
-        switch self {
-            case .not: return true
-            default: return false
-        }
-    }
-}
-
+/// An integer or double constant value parameter (eg `1_000`, `-42.0`)
+///
+/// #TODO
+/// - This is somewhat confusingly named. Possibly would be better to rename as `Numeric`, since
+///   `stringLiteral` *IS* a constant type, or else `stringLiteral` should be moved into this.
 public enum Constant: CustomStringConvertible, Equatable {
     case int(Int)
     case double(Double)
-    
+
     public var description: String {
         switch self {
-            case .int(let i): return i.description
+            case .int(let i):    return i.description
             case .double(let d): return d.description
         }
     }
 }
 
+// MARK:- --- THIS SECTION TO BE MOVED TO A NEW FILE ---
+/// `ParameterDeclaration is NOT used at the lexing stage and is not presentable as a LeafToken - it
+/// is built and interpreted during LeafParser.parse(). Move to a more appropriate file.
+
```

Comment:
Will break this out eventually to go along with LeafParser instead of Lexer since it's not evaluated at this stage of the pipeline